src/Http/Middleware/ProxyAddress.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Http\Middleware;
+
+use Flarum\Http\Exception\ProxyNotAllowedException;
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface as Middleware;
+use Psr\Http\Server\RequestHandlerInterface;
+
+class ProxyAddress implements Middleware
+{
+    /**
+     * @var bool
+     */
+    protected $enabled;
+
+    /**
+     * @var array
+     */
+    protected $allowedAddresses;
+
+    /**
+     * @param bool $enabled
+     * @param array $allowedAddresses
+     */
+    public function __construct($enabled, $allowedAddresses)
+    {
+        $this->enabled = $enabled;
+        $this->allowedAddresses = $allowedAddresses;
+    }
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $ipAddress = Arr::get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
```

Comment:
Can we be confident that reverse proxies would always either set this to 127.0.0.1 (if set at all) or leave it blank? This question also largely comes from inexperience working with reverse proxies, but it would be nice to avoid unnecessary issues.

Reply:
Technically the `REMOTE_ADDR` will never be blank, it will either contain the IP of the client requesting data, or it will contain the IP of the proxy server forwarding the request. It's important that this be the first in the options so that it can be overridden with a correct proxy value if a proxy is used. If this value where blank then there would be a MUCH larger issue with the server.

Reply:
As long as we support some form of wildcard so that if a reverse proxy uses a non-small set of possible options for REMOTE_ADDR we can support that, we should be fine.